typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,25 @@
+const $tabContainer = document.querySelector('.tab-container');
+if (!$tabContainer) throw new Error('$tabContainer not found!');
+$tabContainer.addEventListener('click', clickHandler);
+
+function clickHandler(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  if (eventTarget.matches('.tab')) {
+    for (const tab of document.querySelectorAll('.tab')) {
+      if (tab === eventTarget) {
+        tab.className = 'tab active';
+      } else {
+        tab.className = 'tab';
+      }
+    }
+    for (const view of document.querySelectorAll('.view')) {
+      if (!(view instanceof HTMLElement))
```

Comment:
Same thing here, we would want to query for all of our `.view` elements at the top of the page.